public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderOptions.tsx

```diff
@@ -0,0 +1,107 @@
+  const onQueryTypeChange = (value: LokiQueryType) => {
+    onChange({ ...query, queryType: value });
+    onRunQuery();
+  };
+
+  const onResolutionChange = (option: SelectableValue<number>) => {
+    onChange({ ...query, resolution: option.value });
+  };
+
+  const onLegendFormatChanged = (evt: React.FormEvent<HTMLInputElement>) => {
+    onChange({ ...query, legendFormat: evt.currentTarget.value });
+    onRunQuery();
+  };
+
+  function onMaxLinesChange(e: React.SyntheticEvent<HTMLInputElement>) {
+    const newMaxLines = preprocessMaxLines(e.currentTarget.value);
+    if (query.maxLines !== newMaxLines) {
+      onChange({ ...query, maxLines: newMaxLines });
+    }
+  }
+
+  let queryType = query.queryType ?? (query.instant ? LokiQueryType.Instant : LokiQueryType.Range);
+
+  return (
+    <EditorRow>
+      <QueryOptionGroup title="Options" collapsedInfo={getCollapsedInfo(query, queryType)}>
+        <EditorField
+          label="Legend"
+          tooltip="Series name override or template. Ex. {{hostname}} will be replaced with label value for hostname."
+        >
+          <AutoSizeInput
+            placeholder="{{label}}"
+            id="loki-query-editor-legend-format"
+            type="string"
+            minWidth={14}
+            defaultValue={query.legendFormat}
+            onCommitChange={onLegendFormatChanged}
+          />
+        </EditorField>
+        <EditorField label="Type">
+          <RadioButtonGroup
+            id="options.query.type"
+            options={queryTypeOptions}
+            value={queryType}
+            onChange={onQueryTypeChange}
+          />
+        </EditorField>
+        <EditorField label="Line limit" tooltip="Upper limit for number of log lines returned by query.">
+          <AutoSizeInput
+            className="width-4"
+            placeholder="auto"
+            type="number"
+            min={0}
+            defaultValue={query.maxLines?.toString() ?? ''}
+            onCommitChange={onMaxLinesChange}
+          />
+        </EditorField>
+        <EditorField label="Resolution">
+          <Select
+            isSearchable={false}
+            onChange={onResolutionChange}
+            options={RESOLUTION_OPTIONS}
+            value={query.resolution || 1}
+            aria-label="Select resolution"
+            menuShouldPortal
+          />
+        </EditorField>
+      </QueryOptionGroup>
+    </EditorRow>
+  );
+});
+
+function getCollapsedInfo(query: LokiQuery, queryType: LokiQueryType): string[] {
+  const queryTypeLabel = queryTypeOptions.find((x) => x.value === queryType);
+  const resolutionLabel = RESOLUTION_OPTIONS.find((x) => x.value === (query.resolution ?? 1));
+
+  const items: string[] = [];
+
+  items.push(`Legend: ${getLegendModeLabel(query.legendFormat)}`);
+
+  if (query.resolution) {
+    items.push(`Resolution: ${resolutionLabel?.label}`);
+  }
+
+  items.push(`Type: ${queryTypeLabel?.label}`);
```

Comment:
Line limit/Max lines is missing in collapsed info. I am not sure if we should always show it even if it is default value. And if we should show it if query is metrics query. 

My intuition would say show it always if the query is log query. But then it would make sense to show resolution only if we have metrics query. Btw in datasource.ts we have `isMetricsQuery` function to determine if the query is log query or metrics.